# [nixpkgs-reviewd](https://github.com/phanirithvij/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/phanirithvij/nixpkgs-reviewd/actions/runs/15248146406


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 411030`
Commit: `b670ef635da88d9deb9d81b703b1e8643ac575b8`

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>efibooteditor</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>efibooteditor</summary>
<pre>[ 52%] Building CXX object CMakeFiles/efibooteditor.dir/src/efikeysequence.cpp.o
[ 54%] Building CXX object CMakeFiles/efibooteditor.dir/src/efikeysequenceedit.cpp.o
[ 55%] Building CXX object CMakeFiles/efibooteditor.dir/src/filepathdelegate.cpp.o
[ 57%] Building CXX object CMakeFiles/efibooteditor.dir/src/filepathdialog.cpp.o
[ 58%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkey.cpp.o
[ 60%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeydelegate.cpp.o
[ 61%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeylistmodel.cpp.o
[ 63%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeysdialog.cpp.o
[ 64%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeysview.cpp.o
[ 66%] Building CXX object CMakeFiles/efibooteditor.dir/src/driveinfo.darwin.cpp.o
[ 67%] Building C object CMakeFiles/efibooteditor.dir/src/efivar-lite.c.o
[ 69%] Building C object CMakeFiles/efibooteditor.dir/src/efivar-lite.darwin.c.o
/tmp/nix-build-efibooteditor-1.5.3.drv-0/source/src/efivar-lite.darwin.c:34:20: error: identifier '_get_nvram_variable_name' is reserved because it starts with '_' at global scope [-Werror,-Wreserved-identifier]
   34 | static CFStringRef _get_nvram_variable_name(const efi_guid_t *guid, const char *name)
      |                    ^
1 error generated.
make[2]: *** [CMakeFiles/efibooteditor.dir/build.make:611: CMakeFiles/efibooteditor.dir/src/efivar-lite.darwin.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:97: CMakeFiles/efibooteditor.dir/all] Error 2
make: *** [Makefile:156: all] Error 2</pre>
</details>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 411030`
Commit: `b670ef635da88d9deb9d81b703b1e8643ac575b8`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>efibooteditor</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 411030`
Commit: `b670ef635da88d9deb9d81b703b1e8643ac575b8`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>efibooteditor</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 411030`
Commit: `b670ef635da88d9deb9d81b703b1e8643ac575b8`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>efibooteditor</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>efibooteditor</summary>
<pre>[ 52%] Building CXX object CMakeFiles/efibooteditor.dir/src/efikeysequence.cpp.o
[ 54%] Building CXX object CMakeFiles/efibooteditor.dir/src/efikeysequenceedit.cpp.o
[ 55%] Building CXX object CMakeFiles/efibooteditor.dir/src/filepathdelegate.cpp.o
[ 57%] Building CXX object CMakeFiles/efibooteditor.dir/src/filepathdialog.cpp.o
[ 58%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkey.cpp.o
[ 60%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeydelegate.cpp.o
[ 61%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeylistmodel.cpp.o
[ 63%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeysdialog.cpp.o
[ 64%] Building CXX object CMakeFiles/efibooteditor.dir/src/hotkeysview.cpp.o
[ 66%] Building CXX object CMakeFiles/efibooteditor.dir/src/driveinfo.darwin.cpp.o
[ 67%] Building C object CMakeFiles/efibooteditor.dir/src/efivar-lite.c.o
[ 69%] Building C object CMakeFiles/efibooteditor.dir/src/efivar-lite.darwin.c.o
/tmp/nix-build-efibooteditor-1.5.3.drv-0/source/src/efivar-lite.darwin.c:34:20: error: identifier '_get_nvram_variable_name' is reserved because it starts with '_' at global scope [-Werror,-Wreserved-identifier]
   34 | static CFStringRef _get_nvram_variable_name(const efi_guid_t *guid, const char *name)
      |                    ^
1 error generated.
make[2]: *** [CMakeFiles/efibooteditor.dir/build.make:611: CMakeFiles/efibooteditor.dir/src/efivar-lite.darwin.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:97: CMakeFiles/efibooteditor.dir/all] Error 2
make: *** [Makefile:156: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/ef/efibooteditor/package.nix
+++ b/pkgs/by-name/ef/efibooteditor/package.nix
@@ -1,0 +1,59 @@
+{
+  lib,
+  qt6,
+  zlib,
+  cmake,
+  efivar,
+  pkg-config,
+  fetchFromGitHub,
+  stdenv,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "efibooteditor";
+  version = "1.5.3";
+
+  src = fetchFromGitHub {
+    owner = "Neverous";
+    repo = "efibooteditor";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-xD40ZzkpwerDYC8nzGVqEHLV0KWbxcc0ApquQjrPJTc=";
+  };
+
+  buildInputs = [ zlib ] ++ lib.optional stdenv.hostPlatform.isLinux efivar;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    qt6.qttools
+    qt6.wrapQtAppsHook
+  ];
+
+  postPatch = ''
+    substituteInPlace misc/org.x.efibooteditor.policy \
+      --replace-fail /usr/bin $out/bin
+    substituteInPlace misc/EFIBootEditor.desktop \
+      --replace-fail "1.0" ${finalAttrs.version} \
+      --replace-fail \
+        'pkexec efibooteditor' \
+        'sh -c "pkexec env DISPLAY=$DISPLAY XAUTHORITY=$XAUTHORITY efibooteditor"'
+  '';
+
+  env.BUILD_VERSION = "v${finalAttrs.version}";
+  cmakeBuildType = "MinSizeRel";
+  cmakeFlags = [ "-DQT_VERSION_MAJOR=6" ];
+
+  postInstall = ''
+    install -Dm644 $src/LICENSE.txt $out/share/licenses/${finalAttrs.pname}/LICENSE
+  '';
+
+  meta = {
+    description = "Boot Editor for (U)EFI based systems";
+    homepage = "https://github.com/Neverous/efibooteditor";
+    changelog = "https://github.com/Neverous/efibooteditor/releases/tag/${finalAttrs.src.tag}";
+    license = lib.licenses.lgpl3Plus;
+    platforms = lib.platforms.linux; # TODO build is broken on darwin
+    maintainers = with lib.maintainers; [ phanirithvij ];
+    mainProgram = "efibooteditor";
+  };
+})

